i3ipc.py

```diff
@@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
             else:
                 setattr(self, attr, None)
 
+        # Make sure marks is never None (use []) to simplify client code. Older
+        # versions of i3 used mark instead of marks, so collapse that into
+        # marks to provide a consistent API.
+        if not self.marks:
```

Comment:
By the way, I kind of messed up and included examples/avoid-ws.py in this
pull request. I'm kind of new to pull-requests to be honest, so I didn't
realize that other changes in the same branch automatically got included.
Let me know if you'd like me to remove it.

On Fri, Apr 22, 2016 at 5:04 PM, Tony Crisci <notifications@github.com>
wrote:

> In i3ipc.py
> <https://github.com/acrisci/i3ipc-python/pull/33#discussion_r60817490>:
>
> > @@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
> >              else:
> >                  setattr(self, attr, None)
> >
> > +        # Make sure marks is never None (use []) to simplify client code. Older
> > +        # versions of i3 used mark instead of marks, so collapse that into
> > +        # marks to provide a consistent API.
> > +        if not self.marks:
>
> self.mark (the original) is set to None by the above code and not deleted
> in the new version of the ipc. The Con should rather not have the property
> at all in this case.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/acrisci/i3ipc-python/pull/33/files/2ea9f619a2d16eab83b6f7e81933a7c7449fd338#r60817490>
>